### PR TITLE
Feature/multiworld

### PR DESCRIPTION
**Pull request by Naamloos**
Missing stuff:
- [x] Spawning players back at their last known location
- [x] Spawning players back at their last known world
- [x] Switching worlds prematurely closes the `Joining World...` screen

### CODE REVIEW

File: Obsidian/WorldData/World.cs [hunk 1]
```diff
@@ -24,42 +31,44 @@ public class World : IWorld
...
+        this.Seed = seed ?? throw new ArgumentNullException(nameof(seed));
+
+        this.Generator = (IWorldGenerator)Activator.CreateInstance(generatorType);
+        this.Generator.Init(seed);
     }
 
     public int GetTotalLoadedEntities() => this.Regions.Values.Sum(e => e == null ? 0 : e.Entities.Count);
 
-    public async Task UpdateClientChunksAsync(Client c, bool unloadAll = false)
+    public async Task UpdateClientChunksAsync(Client c, bool unloadAll = false, bool respawning = false)
```
Comment (Jonpro03) [inline Obsidian/WorldData/World.cs]: ~~Is `respawning` ever true?~~ found it
Should this be a property of Client?

File: Obsidian.API/_Interfaces/IConfig.cs [hunk 2]
```diff
@@ -1,5 +1,7 @@
 ﻿namespace Obsidian.API;
 
+// TODO ADD NEW WORLD FORMAT
```
Comment (Naamloos) [inline Obsidian.API/_Interfaces/IConfig.cs]: This still TODO?

File: Obsidian/Commands/MainCommandModule.cs [hunk 3]
```diff
@@ -259,7 +259,7 @@ public async Task GamemodeAsync(CommandContext Context, [Remaining] string args_
     public async Task TeleportAsync(CommandContext Context, [Remaining] VectorF location)
     {
         var player = Context.Player;
-        await player.SendMessageAsync($"ight homie tryna tp you (and sip dicks) {location.X} {location.Y} {location.Z}");
```
Comment (Naamloos) [inline Obsidian/Commands/MainCommandModule.cs:262]: understandable removal

File: Obsidian/Entities/Entity.cs [hunk 4]
```diff
@@ -342,4 +340,9 @@ public bool Equals([AllowNull] Entity other)
     public static bool operator !=(Entity a, Entity b) => !(a == b);
 
     public override int GetHashCode() => this.EntityId.GetHashCode();
+
+    //TODO IMPLEMENT THESE
```
Comment (Naamloos) [inline Obsidian/Entities/Entity.cs]: again, is this still TODO?

File: Obsidian/WorldData/World.cs (hunk 1 above)
Comment (Tides) [reply Obsidian/WorldData/World.cs]: YOu're right I actually didn't think about that I'll make it a property 🙏🏼

File: Obsidian.API/_Interfaces/IConfig.cs (hunk 2 above)
Comment (Tides) [reply Obsidian.API/_Interfaces/IConfig.cs]: I'll remove that ^ ^

File: Obsidian/Entities/Entity.cs (hunk 4 above)
Comment (Tides) [reply Obsidian/Entities/Entity.cs]: Yeah, I'll fully write these out now actually

**Review by Jonpro03 — APPROVED** [review]

**Pull request merged**